3party/opening_hours/opening_hours.cpp

```diff
@@ -0,0 +1,1384 @@
+{
+  if (entry.HasStart())
+    ost << static_cast<uint32_t>(entry.GetStart());
+  if (entry.HasEnd())
+    ost << '-' << static_cast<uint32_t>(entry.GetEnd());
+  return ost;
+}
+
+bool WeekdayRange::HasWday(Weekday const & wday) const
+{
+  if (IsEmpty() || wday == Weekday::None)
+    return false;
+
+  if (!HasEnd())
+    return GetStart() == wday;
+
+  return GetStart() <= wday && wday <= GetEnd();
+}
+
+bool WeekdayRange::HasSunday() const { return HasWday(Weekday::Sunday); }
+bool WeekdayRange::HasMonday() const { return HasWday(Weekday::Monday); }
+bool WeekdayRange::HasTuesday() const { return HasWday(Weekday::Tuesday); }
+bool WeekdayRange::HasWednesday() const { return HasWday(Weekday::Wednesday); }
+bool WeekdayRange::HasThursday() const { return HasWday(Weekday::Thursday); }
+bool WeekdayRange::HasFriday() const { return HasWday(Weekday::Friday); }
+bool WeekdayRange::HasSaturday() const { return HasWday(Weekday::Saturday); }
+
+bool WeekdayRange::HasStart() const
+{
+  return GetStart() != Weekday::None;
+}
+
+bool WeekdayRange::HasEnd() const
+{
+  return GetEnd() != Weekday::None;;
+}
+
+bool WeekdayRange::HasOffset() const
+{
+  return GetOffset() != 0;
+}
+
+bool WeekdayRange::IsEmpty() const
+{
+  return GetStart() == Weekday::None && GetEnd() == Weekday::None;
+}
+
+Weekday WeekdayRange::GetStart() const
+{
+  return m_start;
+}
+
+Weekday WeekdayRange::GetEnd() const
+{
+  return m_end;
+}
+
+size_t WeekdayRange::GetDaysCount() const
+{
+  if (IsEmpty())
+    return 0;
+  return static_cast<uint32_t>(m_start) - static_cast<uint32_t>(m_end) + 1;
+}
+
+void WeekdayRange::SetStart(Weekday const & wday)
+{
+  m_start = wday;
+}
+
+void WeekdayRange::SetEnd(Weekday const & wday)
+{
+  m_end = wday;
+}
+
+int32_t WeekdayRange::GetOffset() const
+{
+  return m_offset;
+}
+
+void WeekdayRange::SetOffset(int32_t const offset)
+{
+  m_offset = offset;
+}
+
+bool WeekdayRange::HasNth() const
+{
+  return !m_nths.empty();
+}
+
+WeekdayRange::TNths const & WeekdayRange::GetNths() const
+{
+  return m_nths;
+}
+
+void WeekdayRange::AddNth(NthWeekdayOfTheMonthEntry const & entry)
+{
+  m_nths.push_back(entry);
+}
+
+std::ostream & operator<<(std::ostream & ost, Weekday const wday)
+{
+  switch (wday)
+  {
+    case Weekday::Sunday:
+      ost << "Su";
+      break;
+    case Weekday::Monday:
+      ost << "Mo";
+      break;
+    case Weekday::Tuesday:
+      ost << "Tu";
+      break;
+    case Weekday::Wednesday:
+      ost << "We";
+      break;
+    case Weekday::Thursday:
+      ost << "Th";
+      break;
+    case Weekday::Friday:
+      ost << "Fr";
+      break;
+    case Weekday::Saturday:
+      ost << "Sa";
+      break;
+    case Weekday::None:
+      ost << "not-a-day";
+  }
+  return ost;
+}
+
+std::ostream & operator<<(std::ostream & ost, WeekdayRange const & range)
+{
+  ost << range.GetStart();
+  if (range.HasEnd())
+    ost << '-' << range.GetEnd();
```

Comment:
When one of the if branches is surrounded by curly braces, the other one should be put in curly braces too.